File: Dave Min/2023-01-31-SearchInRotatedSortedArray.cs
public class Solution
{
    public int Search(int[] nums, int target)
    {
        if (nums.Length == 1)
            return nums[0] == target ? 0 : -1;
        int l = 0, r = nums.Length - 1;
        int startingPoint = 0;

        if (nums[0] < nums[nums.Length - 1])
        {
            var idx = nums.ToList().BinarySearch(target);
            return idx < 0 ? -1 : idx;
        }
        while (l <= r)
        {
            int mid = l + (r - l) / 2;
            if (nums[mid - 1] > nums[mid])
            {
                startingPoint = mid; break;
            }
            if (nums[mid] > nums[mid + 1])
            {
                startingPoint = mid + 1; break;
            }

            if (nums[0] < nums[mid])
            {
                l = mid + 1;
            }
            else
            {
                r = mid - 1;
            }
        }

        Console.WriteLine(startingPoint);
        if (nums[startingPoint] == target)
            return startingPoint;
        else if (nums[startingPoint] < target)
        {
            var list = nums.ToList().GetRange(startingPoint, nums.Length - startingPoint);
            var idx = list.BinarySearch(target);
            return idx < 0 ? -1 : startingPoint + idx;
        }
        else
        {
            var list = nums.ToList().GetRange(0, startingPoint);
            var idx = list.BinarySearch(target);
            return idx < 0 ? -1 : startingPoint + idx;
        }
    }
}

File: Heeseop An/2023-01-31-Search-in-Rotated-Sorted-Array.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left_index = 0
        right_index = len(nums) - 1

        while left_index <= right_index:
            mid_index = left_index + (right_index - left_index) // 2
            if nums[mid_index] == target:
                return mid_index
            elif nums[left_index] <= nums[mid_index]:
                if target >= nums[left_index] and target < nums[mid_index]:
                    right_index = mid_index
                else:
                    left_index = mid_index + 1
            else:
                if target > nums[mid_index] and target <= nums[right_index]:
                    left_index = mid_index + 1
                else:
                    right_index = mid_index
        return -1
